From ff7a20c7b3f697c2eef50810d4532b5071ade530 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Wed, 27 Dec 2023 14:24:36 +0100
Subject: [PATCH 7/9] arm64: dts: ti: k3-am64-main: icssg1_mdio: fix clock
 reference

icssg1_mdio should use the same clock that is used by icssg1 core.
"icssg1_coreclk_mux" already selects iclk ("&k3_clks 82 20") over
core_clk ("&k3_clks 82 0").

When icssg1_mdio selects core_clk and is probed before
icssg1_coreclk_mux, a deadlock occurs because the mux clock can never
probe. This lead to cascading failure probing pru netdevs.

Update icssg1_mdio clock reference to iclk.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 arch/arm64/boot/dts/ti/k3-am64-main.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/ti/k3-am64-main.dtsi b/arch/arm64/boot/dts/ti/k3-am64-main.dtsi
index fd71284e7c7d..6f9c3ae07632 100644
--- a/arch/arm64/boot/dts/ti/k3-am64-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am64-main.dtsi
@@ -1423,7 +1423,7 @@ icssg1_mdio: mdio@32400 {
 			reg = <0x32400 0x100>;
 			#address-cells = <1>;
 			#size-cells = <0>;
-			clocks = <&k3_clks 82 0>;
+			clocks = <&k3_clks 82 20>;
 			clock-names = "fck";
 			bus_freq = <1000000>;
 			status = "disabled";
-- 
2.35.3

